source/shared/cpp/ObjectModel/ParseUtil.cpp

```diff
@@ -334,6 +338,25 @@ std::vector<std::shared_ptr<BaseCardElement>> ParseUtil::GetElementCollection(
         {
             // Use the parser that maps to the type
             elements.push_back(parser->Deserialize(elementParserRegistration, actionParserRegistration, curJsonValue));
+            if (elements.back()->GetElementType() == CardElementType::TextBlock)
+            {
+                auto textBlock = std::static_pointer_cast<TextBlock>(elements.back()); 
+                if (textBlock != nullptr)
+                {
+                    textBlock->SetLanguage(dateLanguage);
+                }
+            }
+            else
+            {
+                if (elements.back()->GetElementType() == CardElementType::Container)
+                {
+                    auto container = std::static_pointer_cast<Container>(elements.back());
+                    if (container != nullptr)
+                    {
+                        container->SetLanguage(dateLanguage);
+                    }
+                }
+            }
```

Comment:
This is the same as the code in Container.cpp, can we make this chunk of code a helper function?

Reply:
Alternatively, instead of doing this here, how about just setting the language on the card at the end, and letting the SetLanguage() methods handle propagating to the elements?

---
In reply to: [170750454](https://github.com/Microsoft/AdaptiveCards/pull/1102#discussion_r170750454) [](ancestors = 170750454)